### ISSUE TITLE
The "version" configuration option is not clearly documented.

### ISSUE THREAD

**Issue by lddubeau** [localForage/localForage#658]
The documentation for the `version` configuration option currently states:

> version
> The version of your database. May be used for upgrades in the future; currently unused.

The problem is that this does not clarify whether this value is something that the client code may use. "May be used for upgrades in the future" does not clarify because it could mean that eventually localForage may have facilities to detect and notify client code when localForage finds an old version of the database that the client code should upgrade. Actually #117 shows that such an idea has been floated in the past. A user following this interpretation may decide to fill the field with whatever version number suits them with the idea that they'll (somehow) handle the version changes themselves for now until localForage provides some sort of formal support.

However, upon reading the code, I see localForage has its own `onupgradeneeded` handler where it may do perform actions internal to localForage. So it seems to me that this number is something that should be used *only* if localForage's documentation says to use another value. It should be reserved for upgrades *of localForage's internal structures*, and not for upgrades to the structures accessible to the client code. I would not call it "unused". It is used to set the database version as understood by IndexedDB (and WebSQL, too I assume). If it were unused, I'd expect that setting it to 2, 30, or 1243 would not matter, but people setting it to such values may run into trouble later when localForage's documentation eventually states 

> to take advantage of localForage's new feature X, your database must be upgraded to version 2.

The poor dev who has already set it to 2 will find himself unable to upgrade, and those who are at higher values will get a warning about a downgrade and probably erratic behavior.

(As a aside, an additional thing that makes me question the value of this `version` option is that it seems the `localStorage` backend simply tosses the value aside. When I look into a `localStorage` instance populated by localforage, I cannot see the version value stored anywhere. Did I miss something?)

Even if my inferences are not correct above, I'd still say the role this option may play in the future is so nebulous right now that until its role is crystal-clear the documentation should not say that it is unused but instead tell users to not use it. (Or remove it from the options??)

**Comment by paulpooch** [comment]
If you accidentally pass a semver style string to the version option.  All of local forage just silently fails.  Very confusing.

**Comment by thgreasi** [comment]
I would expect that .ready() would reject in that case. Would you prefer if
it threw an error right away?

> --
*Thodoris Greasidis*

**Comment by paulpooch** [comment]
It did not reject.  I think the promise just hangs and never resolves.  No
catch blocks detected any issue.

I mean it's not a huge deal just maybe clarify the docs a little.  Looks to
me like indexedDB just runs on int versioning... It took me an hour of
wasted time to figure that out.

**Comment by thgreasi** [comment]
I will change it to do something more meaningful asap.

**Comment by augbog** [comment]
Just curious what the status of this is?

**Comment by tofumatt** [comment]
We haven't been able to get to it as we've been a bit busy fixing other issues, but we would love a pull request. If not I'll try to get to it later this month.

**Comment by augbog** [comment]
@tofumatt Thanks for the quick reply! I really appreciate all the hard work :) may try to help if I can with documentation parts (assuming I understand them correctly :P)

**Comment by tofumatt** [comment]
Certainly! I haven’t looked at the issue and what it involves so I forget it a bit :-)

But if you understand the issue and want to send a PR I’ll be sure to review it ASAP, thanks! :-)
